src/cmd/services/m3query/config/config.go

```diff
@@ -323,13 +327,15 @@ func (l *PerQueryLimitsConfiguration) AsLimitManagerOptions() cost.LimitManagerO
 func (l *PerQueryLimitsConfiguration) AsFetchOptionsBuilderOptions() handleroptions.FetchOptionsBuilderOptions {
 	if l.MaxFetchedSeries <= 0 {
```

Comment:
Think we should treat max fetched series and max fetched docs independent here. It's strange that docs limit gets the default even if it had a value specified in config, only because max fetch series in config wasn't set (i.e. maxfetchedseries <= 0).

i.e.

```go
seriesLimit := defaultStorageQuerySeriesLimit
if v := l.MaxFetchedSeries; v > 0 {
  seriesLimit = v
}

docsLimit := defaultStorageQueryDocsLimit
if v := l.MaxFetchedDocs; v > 0 {
  docsLimit = v
}

return handleroptions.FetchOptionsBuilderOptions{
  SeriesLimit:       int(seriesLimit),
  DocsLimit:       int(docsLimit),
  RequireExhaustive:       int(l.RequireExhaustive),
}

```